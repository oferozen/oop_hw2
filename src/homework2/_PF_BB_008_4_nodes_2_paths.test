#syntax:
# CreateNode n w
# CreateGraph n
# AddNode g n
# AddEdge A n3 n1 
# FindPath g n1 n2 -> n3 n4

CreateGraph g
CreateNode n1 1
CreateNode n2 2
CreateNode n3 1
CreateNode n4 1
AddNode g n1
AddNode g n2
AddNode g n3
AddNode g n4

# n4 is the source and n1 is the goald.

AddEdge g n4 n3
AddEdge g n4 n2
AddEdge g n3 n1
AddEdge g n2 n1

FindPath g n4 -> n1
FindPath g n1 -> n4
FindPath g n3 -> n1
FindPath g n2 -> n1

# Add a path with weight 3
CreateNode n5 2
CreateNode n6 0
AddNode g n5
AddNode g n6
AddEdge g n4 n5
AddEdge g n5 n6
AddEdge g n6 n1
FindPath g n4 -> n1
