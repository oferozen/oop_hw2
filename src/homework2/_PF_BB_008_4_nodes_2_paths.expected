#syntax:
# CreateNode n w
# CreateGraph n
# AddNode g n
# AddEdge A n3 n1 
# FindPath g n1 n2 -> n3 n4

created graph g
created node n1 with cost 1
created node n2 with cost 2
created node n3 with cost 1
created node n4 with cost 1
added node n1 to g
added node n2 to g
added node n3 to g
added node n4 to g

# n4 is the source and n1 is the goald.

added edge from n4 to n3 in g
added edge from n4 to n2 in g
added edge from n3 to n1 in g
added edge from n2 to n1 in g

shortest path in g: n4 n3 n1
shortest path in g:
shortest path in g: n3 n1
shortest path in g: n2 n1

# Add a path with weight 3
created node n5 with cost 2
created node n6 with cost 0
added node n5 to g
added node n6 to g
added edge from n4 to n5 in g
added edge from n5 to n6 in g
added edge from n6 to n1 in g
shortest path in g: n4 n3 n1
